gotpm/cmd/seal_test.go

```diff
@@ -84,17 +95,23 @@ func TestUnsealFail(t *testing.T) {
 	rwc := internal.GetTPM(t)
 	defer tpm2tools.CheckedClose(t, rwc)
 	ExternalTPM = rwc
+	extendPCR := func(rw io.ReadWriter, pcr int) error {
+		pcrToExtend := tpmutil.Handle(pcr)
+		extension := bytes.Repeat([]byte{0xAA}, sha256.Size)
+		return tpm2.PCRExtend(rw, pcrToExtend, tpm2.AlgSHA256, extension, "")
+	}
 
 	tests := []struct {
-		name    string
-		tpmFunc func(io.ReadWriter) error
+		name             string
+		sealPCRs         string
+		certifyPCRs      string
+		tpmFunc          func(io.ReadWriter, int) error
```

Comment:
Something feels off about expanding this with an additional parameter.

I would remove the tpmFunc parameter, and just make a helper for extendPCR. If Joe (or someone else) wants to make a parameter so we can call a tpmFunc in between seal/unseal to resolve that TODO, they can add back the tpmFunc parameter.

Reply:
The problem is we might not need `extendPCR` in every test cases in the future. We need a way to execute some (0 to n) functions between seal/unseal.